browser/src/Services/Themes/ThemeManager.ts

```diff
@@ -141,11 +153,78 @@ export const getHoverColors = (
     return userHoverColors
 }
 
-export const getColorsFromConfig = (config: Configuration, colors: Partial<IThemeColors>) => {
-    const userConfig = config.getValues()
-    const hoverColors = getHoverColors(userConfig, colors)
+const updateThemeWithDefaults = (tokenColors: Partial<IThemeColors>) => {
+    const updatedTheme = Object.keys(tokenColors).reduce((acc, t) => {
+        const item = tokenColors[t]
+        if (item && !item.settings.foreground && item.settings.fallback) {
+            acc[t] = {
+                ...item,
+                settings: {
+                    ...item.settings,
+                    ...tokenColors[item.settings.fallback.toLowerCase()].settings,
+                },
+            }
+            return acc
+        }
+        return acc
+    }, tokenColors)
+    return { "editor.tokenColors": updatedTheme }
+}
+
+const getTokenColors = ({
+    config,
+    defaultTokens,
+    themeColors,
+    vimColors,
+}: {
+    config: Configuration
+    defaultTokens: IEditorTokens
+    themeColors: IEditorTokens
+    vimColors: { [token: string]: ITokenColorsSetting }
+}) => {
+    const userTokens = config.getValue("editor.tokenColors")
```

Comment:
One thing that is a little bit confusing here is that the `ThemeManager` is operating at a different level of abstraction for `tokenColors` vs `colors`.

For `colors`, the `ThemeManager` is mainly responsible for loading the colors and getting the defaults. The `Colors` calls in `Services/Colors` brings together the theme colors with the user's configuration, and handles the merging (it's basically a layer on top, that coalesces it with the configuration). It's main responsibility is listening for both the `ThemeManager.onThemeChanged` and `configuration.onConfigurationChanged` to pick up color changes in either place.

For `tokenColors`, the `ThemeManager` is doing the merging of the user configuration, which is more responsibility than it has with `colors`. I could see us later wanting to have some similiar logic as for `colors` - where we listen to configuration changes and update them. It might be worth decoupling that / abstracting that like we have for `colors`, at some point.